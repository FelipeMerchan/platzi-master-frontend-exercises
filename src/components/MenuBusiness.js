import React, { useState, useEffect } from 'react'

import './styles/MenuBusiness.css'

function MenuBusiness(props) {
  const [count, setCount] = useState(0)
  const composeEvent = new Event('composed')

  const handleClick = event => {
    event.preventDefault()
    props.onClick(event)
    const { target: { id } } = event

    event.target.dispatchEvent(composeEvent);
    setCount(count + 1)
    console.log(`Id:${id}, Count: ${count}`)
  }

  useEffect(() => {
    const links = document.querySelectorAll('.MenuBusiness__item > a')

    links.forEach(object => {
      object.addEventListener('composed', function(event) {
        event.stopPropagation()
        console.log('IÂ´m not propagating')
      })
    })
  })

  return (
    <nav className="MenuBusiness">
      <ul>
        <li className="MenuBusiness__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button='nav' href="#home" id="home" >Inicio</a>
        </li>
        <li className="MenuBusiness__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button='nav' href="#aboutUs" id="aboutUs" >Acerca</a>
        </li>
        <li className="MenuBusiness__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button='nav' href="#services" id="services" >Servicios</a>
        </li>
        <li className="MenuBusiness__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button='nav' href="#products" id="products" >Productos</a>
        </li>
        <li className="MenuBusiness__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button='nav' href="#contact" id="contact" >Contacto</a>
        </li>
      </ul>
    </nav>
  )
}

export default MenuBusiness