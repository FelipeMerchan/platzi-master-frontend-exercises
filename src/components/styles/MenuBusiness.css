.MenuBusiness,
.MenuBusiness ul {
  display: inherit;
  height: inherit;
}

.MenuBusiness {
  align-self: flex-start;
}

.MenuBusiness__item {
  height: inherit;
  margin-right: var(--space-2);
  padding: calc(var(--space) / 2);
  list-style-type: none;
}

.MenuBusiness__item:last-child {
  margin-right: 0;
}

.MenuBusiness__item a {
  padding: calc(var(--space) / 2) var(--space-2);
  color: #a6a7a2;
  text-decoration: none;
}

.MenuBusiness__item a:hover {
  color: #ffffff;
  background-color: var(--actionColor);
}