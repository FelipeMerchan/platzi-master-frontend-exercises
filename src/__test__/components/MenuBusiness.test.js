import React from 'react'
import { mount, shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import MenuBusiness from '../../components/MenuBusiness'

describe('Probar el componente <MenuBusiness />', () => {
  const menuBusiness = shallow(<MenuBusiness />)
  test('Render del componente MenuBusiness', () => {
    expect(menuBusiness.length).toEqual(1)
  })
  test('Render con el className MenuBusiness', () => {
    expect(menuBusiness.find('nav').hasClass('MenuBusiness')).toBe(true)
  })

  test('Render de los 5 elementos de lista del menú', () => {
    expect(menuBusiness.find('ul').children()).toHaveLength(5);
  });
})

describe('Probar los comportamientos del componente MenuBusiness', () => {
  const onClick = jest.fn()
  const menuBusiness = mount(<MenuBusiness onClick={onClick} />, { attachTo: document.body })
  const firstButton = menuBusiness.find('#home')
  test('Comprobar que los botones del footer cambien de color a través de los clics de los botones de navegación', () => {
    firstButton.simulate('click')
    expect(onClick).toHaveBeenCalledTimes(1);
  });
})

describe('MenuBusiness Snapshot', () => {
  test('Comprobar la UI del componente MenuBusiness', () => {
    const menuBusiness = create(<MenuBusiness />)
    expect(menuBusiness.toJSON()).toMatchSnapshot()
  })
})