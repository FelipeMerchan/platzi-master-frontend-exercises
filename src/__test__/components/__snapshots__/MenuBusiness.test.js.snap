// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBusiness Snapshot Comprobar la UI del componente MenuBusiness 1`] = `
<nav
  className="MenuBusiness"
>
  <ul>
    <li
      className="MenuBusiness__item"
    >
      <a
        data-button="nav"
        href="#home"
        id="home"
        onClick={[Function]}
        role="button"
      >
        Inicio
      </a>
    </li>
    <li
      className="MenuBusiness__item"
    >
      <a
        data-button="nav"
        href="#aboutUs"
        id="aboutUs"
        onClick={[Function]}
        role="button"
      >
        Acerca
      </a>
    </li>
    <li
      className="MenuBusiness__item"
    >
      <a
        data-button="nav"
        href="#services"
        id="services"
        onClick={[Function]}
        role="button"
      >
        Servicios
      </a>
    </li>
    <li
      className="MenuBusiness__item"
    >
      <a
        data-button="nav"
        href="#products"
        id="products"
        onClick={[Function]}
        role="button"
      >
        Productos
      </a>
    </li>
    <li
      className="MenuBusiness__item"
    >
      <a
        data-button="nav"
        href="#contact"
        id="contact"
        onClick={[Function]}
        role="button"
      >
        Contacto
      </a>
    </li>
  </ul>
</nav>
`;
